plugins/git4idea/src/git4idea/checkin/GitCheckinEnvironment.java

```diff
@@ -107,6 +107,34 @@ public RefreshableOnComponent createAdditionalOptionsPanel(CheckinProjectPanel p
     return new GitCheckinOptions(myProject, panel);
   }
 
+  @Nullable
+  public Set<String> getCommitTemplatesFor(FilePath[] filesToCheckin) {
+    LinkedHashSet<String> templates = ContainerUtil.newLinkedHashSet();
+    for (VirtualFile root : GitUtil.gitRoots(Arrays.asList(filesToCheckin))) {
+      try {
+        final String commitTemplateFile = GitConfigUtil.getValue(myProject, root, "commit.template");
+        if (commitTemplateFile != null && commitTemplateFile.length() > 0) {
+          final String encoding = GitConfigUtil.getCommitEncoding(myProject, root);
+          final String templateContent = FileUtil.loadFile(new File(commitTemplateFile), encoding);
+          if (templateContent.length() > 0) {
+            templates.add(templateContent);
+          }
+        }
+      }
+      catch (VcsException e) {
+        if (log.isDebugEnabled()) {
```

Comment:
Is the check "isDebugEnabled" already tested in the log.debug() method ?

It seems redundant.

Reply:
but it allows not to create the message, and not to call the root.getPath() method.
so a small optimisation.